### PR TITLE
Pause game and mute music when game window loses focus

### PR DESCRIPTION
**Pull request by JoBergeron**
When playing singleplayer, when the game window loses focus, I thought it would be nice to mute the music and pause the game.

This is very useful when you're playing for 10-15 at home and a pesky Teams meeting springs up and you just want to answer without having the music playing in the back.

### CODE REVIEW

File: Source/diablo.cpp [hunk 1]
```diff
@@ -1648,10 +1648,41 @@ void diablo_pause_game()
...
+			if (!gameWasAlreadyPaused) {
+				PauseMode = 0;
+			}
+
+			music_unmute();
+			minimizePaused = false;
+		}
+		
+		//force_redraw = 255;
+	}
```
Comment (AJenbo) [inline Source/diablo.cpp:1683]:
```suggestion
	if (gbIsMultiplayer) {
		return;
	}

	if (!minimizePaused) {
		gameWasAlreadyPaused = PauseMode != 0;

		if (!gameWasAlreadyPaused) {
			PauseMode = 2;
			sound_stop();
			track_repeat_walk(false);
		}

		music_mute();
		minimizePaused = true;
	} else {
		if (!gameWasAlreadyPaused) {
			PauseMode = 0;
		}

		music_unmute();
		minimizePaused = false;
	}
```

File: Source/diablo.cpp [hunk 2]
```diff
@@ -1648,10 +1648,41 @@ void diablo_pause_game()
 			sound_stop();
 			track_repeat_walk(false);
 		}
+
 		force_redraw = 255;
 	}
 }
 
+bool gameWasAlreadyPaused = false;
+bool minimizePaused = false;
```
Comment (AJenbo) [inline Source/diablo.cpp:1657]:
```suggestion
bool GameWasAlreadyPaused = false;
bool MinimizePaused = false;
```

**Comment by AJenbo** [comment]
I can see this generally being useful, especially on phones.

**Review by AJenbo — APPROVED** [review]

**Comment by qndel** [comment]
Love this btw :D

File: Source/sound.cpp [hunk 3]
```diff
@@ -306,4 +306,16 @@ int sound_get_or_set_sound_volume(int volume)
 	return sgOptions.Audio.nSoundVolume;
 }
 
+void music_mute()
+{
+	if (music)
```
Comment (julealgon) [inline Source/sound.cpp:311]: @AJenbo weren't we using explicit checks?

Comment (AJenbo) [reply Source/sound.cpp:311]: yeah, not sure what type this is, but clang-tidy should get it if it's not a bool

**Pull request merged**